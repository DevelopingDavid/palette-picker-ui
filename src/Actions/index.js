export const addPalette = (palette) => ({
  type: 'ADD_PALETTE',
  palette
})

export const toggleLocked = (color) => ({
  type: 'TOGGLE_LOCKED',
  color
})

export const makeNewProject = (project) => ({
  type: "MAKE_NEW_PROJECT",
  project
})

export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
})

export const hasError = (message) => ({
  type: "HAS_ERROR",
  message
})