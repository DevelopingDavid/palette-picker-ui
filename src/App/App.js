import React, { Component } from 'react';
import HeaderBar from '../HeaderBar/HeaderBar';

class App extends Component {
  render () {
    return (
      <div className="app">
        <HeaderBar/>
      </div>
    );
  }
}

export default App;
