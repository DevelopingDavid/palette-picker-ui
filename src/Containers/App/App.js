import React, { Component } from 'react';
import HeaderBar from '../../HeaderBar/HeaderBar';
import CardContainer from '../../Components/CardContainer/CardContainer';

class App extends Component {
  constructor() {
    super()
    this.state = {
      projects: []
    }
  } 

  componentDidMount() {
    this.generateColors()
  }

  generateColors = () => {
    let projects = [];
    for (var i = 0; i < 5; i++) {
      let color = '#' + Math.random().toString(16).slice(2, 8).toUpperCase()
      projects.push(color)
    }
    this.setState({ projects }) 
  }

  render () {
    return (
      <div className="app">
        <HeaderBar />
        <CardContainer 
          generateColors={this.generateColors}
          projects={this.state.projects} />
      </div>
    );
  }
}

export default App;
