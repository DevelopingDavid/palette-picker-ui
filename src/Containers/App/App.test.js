import React from 'react';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { addPalette } from '../../Actions'
jest.mock('../../Actions')
import { createShallow } from '@material-ui/core/test-utils';

const mockPalette = [
  {
    hex: "#ffff",
    locked: false
  },
  {
    hex: "#fff",
    locked: false
  }, {
    hex: "#fffF68",
    locked: false
  }, {
    hex: "#fff",
    locked: false
  }, {
    hex: "#3C4F68",
    locked: false
  }
]

describe('App', () => {
  let wrapper;
  let shallow;

  beforeEach(() => {
    shallow = createShallow();
    wrapper = shallow(
      <App currentPalette={mockPalette} addPalette={jest.fn()} />
      // <Project project={mockProject} fetchProjectsThunk={mockFunction} />
    )
  });
  it('should match Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should mapStateToProps', () => {
      const mockState = {}
      const upDatedState = {
        currentPalette: []
      }
      const firstState = mapStateToProps(mockState)
      expect(firstState).toEqual({})
      const newState = mapStateToProps(upDatedState)
      expect(newState).toEqual(upDatedState)
    
    })

    it('should dispatch addPalette', () => {
      const mockDispatch = jest.fn()
      const mockPalette = {
        project_name: 'a;lskdjf',
        color_one: '#fff',
        color_two: '#aaa',
        color_three: '#bbb',
        color_four: '#aaa',
        color_five: '#fff'
      }
      const actionToDispatch = addPalette(mockPalette)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.addPalette(mockPalette)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})