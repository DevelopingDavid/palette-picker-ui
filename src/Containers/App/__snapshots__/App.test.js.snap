// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match Snapshot 1`] = `
<div
  className="app"
>
  <ConnectFunction />
  <ConnectFunction
    checkLockedColors={[Function]}
  />
</div>
`;
